fhf-datafangst-client/src/components/Benchmark/BenchmarkCards.tsx

```diff
@@ -45,71 +48,143 @@ enum BenchmarkType {
   FishingWeight,
 }
 
+const createDataset = (
+  ownTrips: Trip[],
+  followTrips: Record<number, Trip[]>,
+  dataExtractor: (t: Trip) => number,
+  reducer: (d: number) => number,
+) => {
+  const profile = useAppSelector(selectBwUserProfile);
+  const vesselInfo = profile?.vesselInfo;
+  const vessels = useAppSelector(selectVesselsByCallsign);
+  const fiskeridirVessels = useAppSelector(selectVesselsByFiskeridirId);
+  const vessel = vesselInfo?.ircs ? vessels[vesselInfo.ircs] : undefined;
+
+  const vesselNames = [vessel?.fiskeridir.name ?? ""];
+  const dataset = [
+    ["date", "vesselName", "value"],
+    ...ownTrips.map((trip) => [
+      trip.start,
+      vessel?.fiskeridir.name ?? "",
+      reducer(dataExtractor(trip)),
+    ]),
+    ...Object.keys(followTrips).reduce(
+      (acc: [string, string, number][], fiskeridirId: string) => {
+        const name = fiskeridirVessels[+fiskeridirId].fiskeridir.name ?? "";
+        vesselNames.push(name);
+        const tmp: [string, string, number][] = followTrips[+fiskeridirId].map(
+          (trip) => [trip.start, name, reducer(dataExtractor(trip))],
+        );
+        return [...acc, ...tmp];
+      },
+      [],
+    ),
+  ];
+  return { dataset, vesselNames };
+};
+
 export const BenchmarkCards: FC = () => {
   const dispatch = useAppDispatch();
+  const followTrips = useAppSelector(selectBenchmarkTrips);
+
+  const followTripsList = Object.values(followTrips).reduce(
```

Comment:
const followTripsList = Object.values(followTrips).flat(); will produce the same result, no? This is the second time I see this, perhaps consider making an own selector in the redux store for this?

Reply:
Looks like it does, didnt know about that method. Thanks!

Reply:
As for putting it into the redux state i feel that would make the code more bloaty, its only used two times as it is and it would make the redux state store two copies of the same data.